File: services/mono/Bacera.Gateway.Core/Models/Tenant/Deposit.cs
using System;
using System.Collections.Generic;

namespace Bacera.Gateway;

public partial class Deposit
{
    public long Id { get; set; }

    public int Type { get; set; }

    public long PartyId { get; set; }

    public long PaymentId { get; set; }

    public int FundType { get; set; }

    public int CurrencyId { get; set; }

    public long Amount { get; set; }

    public long? TargetAccountId { get; set; }

    public string ReferenceNumber { get; set; } = null!;

    public virtual Currency Currency { get; set; } = null!;

    public virtual FundType FundTypeNavigation { get; set; } = null!;

    public virtual Matter IdNavigation { get; set; } = null!;

    public virtual Party Party { get; set; } = null!;

    public virtual Payment Payment { get; set; } = null!;

    public virtual Account? TargetAccount { get; set; }
}

File: app/web/vue/src/projects/client/modules/wallet/models/Deposit.ts
import { Matter } from "@/core/models/Matter";
import { ICriteria } from "@/core/models/Criteria";

export interface Deposit {
  id: number;
  matterId: number;
  type: number;
  partyId: number;
  paymentId: number | null;
  currencyId: number;
  amount: number;
  matter: Matter | null;
}
export interface DepositCriteria extends ICriteria {
  partyId: number | null;
  matterId: number | null;
  from: string | null;
  to: string | null;
  stateId: number | null;
  currencyId: number | null;
}
